feature_engine/encoding/decision_tree.py

```diff
@@ -239,6 +281,15 @@ def fit(self, X: pd.DataFrame, y: pd.Series):
         variables_ = self._check_or_select_variables(X)
         _check_contains_na(X, variables_)
 
+        # dictionary of categories and seen values
+        if self.unseen in ["encode", "ignore"]:
+            self._categories = {}
+            for var in variables_:
+                unique_vars = X[var].drop_duplicates().to_list()
```

Comment:
is this trying to capture the unique categories per variable? can we not do X[var].unique(() instead?

the name unique_vars is confusing, should it not be unique_categories? or simply categories?